bors.toml

```diff
@@ -23,7 +23,6 @@ status = [
     "check-msrv",
     "slasher-tests",
     "syncing-simulator-ubuntu",
-    "disallowed-from-async-lint",
```

Comment:
Any particular reason for removing this?

Reply:
The lint is written against an old compiler version, so updating the PINNED_NIGHTLY breaks it. It would also break as soon as we bump the MSRV to 1.65 (which we'll need for Capella).

I had a brief go at rewriting the lint to work with Rust 1.65+ but I ran into ICEs that I didn't understand. I've recommended to Diva that she delete it in this PR and we'll revive the lint at some point in the future. Hopefully using https://github.com/trailofbits/dylint.